### DISCUSSION TITLE
主成分PCA提取

### DISCUSSION THREAD

**Discussion by zhaoliang0302** [ShixiangWang/self-study#44] [Q&A] [answered]
翔哥好，
我看文章时看到作者描述根据基因表达矩阵鉴定的module后和临床形状（连续变量）相关联，其中根据PCA挑出来的主成分步骤不知道如何实现
> To relate each cluster with clinical outcomes, principal component analysis (PCA) is first performed for each cluster and then the correlation between the first (or multiple) principal component(s) and each trait is computed as cluster relevance to the trait

这里面的first (or multiple) principal component(s)怎么去界定呢？因为有时候一个成分不能很好地代表，但是文章没说。如果挑出来的几个基因怎么合并成一个值和性状相关性分析？
因为鉴定出的module有很多，如果一步步这么选的话不实际，但是这个步骤循环怎么写我也不知道。能否给点指点？谢谢

这篇文章是一个类似WGCNA的包，doi: 10.1371/journal.pcbi.1004574

**Comment by zhaoliang0302** [comment]
我问了作者都没人回复我:rofl:，这个包官方文档也没写，来求助大神了

**Comment by ShixiangWang** [comment, marked as answer]
一般情况下你可以选权重最大的主成分，另外也可以写程序选择累积 >50% 的前几个主成分。

**Comment by zhaoliang0302** [comment]
谢谢，我需要好好琢磨一下
